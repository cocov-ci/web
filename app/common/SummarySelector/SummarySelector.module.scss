@import "styles/variables";

.base {
  display: flex;
  border: 1px solid $colorGray;
  border-radius: 8px;
  justify-content: space-between;

  &.gutterBottom {
    margin-bottom: 32px;
  }

  .textContainer {
    padding: 20px;
    display: flex;
    align-items: center;

    .disclaimer {
      margin-left: 14px;
    }

    .branchName {
      font-variation-settings: "wght" 600;
    }
  }

  .buttonContainer {
    margin-right: 25px;
    display: flex;
    align-items: center;
  }
}
