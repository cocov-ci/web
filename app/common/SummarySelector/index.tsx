'use client'

import classNames from 'classnames'
import { GitBranch } from 'lucide-react'
import React, { useState } from 'react'

import BranchSwitcher from 'app/common/BranchSwitcher'
import Button from 'app/common/Button'
import { inconsolata } from 'utils/fonts'

import styles from './SummarySelector.module.scss'

type SummarySelectorProps = {
  className?: string
  gutterBottom?: boolean
  branchName: string
  branchesList: string[]
}

const SummarySelector = ({
  className,
  branchName,
  gutterBottom,
}: SummarySelectorProps) => {
  const [showBranchSwitcher, setShowBranchSwitcher] = useState<boolean>(false)

  return (
    <div
      className={classNames(styles.base, className, {
        [styles.gutterBottom]: gutterBottom,
      })}
    >
      <div className={styles.textContainer}>
        <GitBranch size={18} />
        <span className={styles.disclaimer}>
          This is a summary of the{' '}
          <span
            className={classNames(styles.branchName, inconsolata.className)}
          >
            {branchName}
          </span>{' '}
          branch.
        </span>
      </div>
      <div className={styles.buttonContainer}>
        <div className={styles.buttonWrapper}>
          <Button
            onClick={() => setShowBranchSwitcher(!showBranchSwitcher)}
            style="mini"
          >
            Switch Branch
          </Button>
          <BranchSwitcher
            className={styles.branchSwitcher}
            onClose={() => setShowBranchSwitcher(false)}
            visible={showBranchSwitcher}
          />
        </div>
      </div>
    </div>
  )
}

export default SummarySelector
