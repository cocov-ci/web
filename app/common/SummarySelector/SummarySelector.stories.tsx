import type { Meta, StoryObj } from '@storybook/react'

import SummarySelector from './index'

const meta: Meta<typeof SummarySelector> = {
  title: 'Common/SummarySelector',
  component: SummarySelector,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof SummarySelector>

export const Default: Story = {
  args: {
    branchName: 'master',
  },
}

export const GutterBottom: Story = {
  args: {
    branchName: 'master',
    gutterBottom: true,
  },
}
