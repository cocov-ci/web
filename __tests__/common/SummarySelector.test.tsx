import { fireEvent, render } from '@testing-library/react'

import SummarySelector from 'app/common/SummarySelector'

import '@testing-library/jest-dom'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
  })),
}))

describe('common/SummarySelector', () => {
  it('renders SummarySelector', () => {
    const { container } = render(
      <SummarySelector branchName="master" branchesList={[]} />,
    )

    expect(
      container.querySelector('.buttonWrapper > button'),
    ).toHaveTextContent('Switch Branch')
  })

  it('triggers Switch Branch button', () => {
    const { container } = render(
      <SummarySelector branchName="master" branchesList={[]} />,
    )

    fireEvent.click(
      container.querySelector('.buttonWrapper > button') as Element,
    )

    expect(container.querySelector('.branchSwitcher')).toBeVisible()
  })

  it('triggers close BranchSwitcher', () => {
    const { container } = render(
      <SummarySelector branchName="master" branchesList={[]} />,
    )

    fireEvent.click(
      container.querySelector('.buttonWrapper > button') as Element,
    )

    expect(container.querySelector('.branchSwitcher.visible')).not.toBeNull()

    fireEvent.click(
      container.querySelector('.branchSwitcher button') as Element,
    )

    expect(container.querySelector('.branchSwitcher.visible')).toBeNull()
  })

  it('renders SummarySelector snapshots', () => {
    const { container } = render(
      <SummarySelector branchName="master" branchesList={[]} />,
    )

    expect(container).toMatchSnapshot()
  })
})
