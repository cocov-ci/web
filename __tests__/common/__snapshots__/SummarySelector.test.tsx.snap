// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/SummarySelector renders SummarySelector snapshots 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="textContainer"
    >
      <svg
        class="lucide lucide-git-branch"
        fill="none"
        height="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="6"
          x2="6"
          y1="3"
          y2="15"
        />
        <circle
          cx="18"
          cy="6"
          r="3"
        />
        <circle
          cx="6"
          cy="18"
          r="3"
        />
        <path
          d="M18 9a9 9 0 0 1-9 9"
        />
      </svg>
      <span
        class="disclaimer"
      >
        This is a summary of the
         
        <span
          class="branchName className"
        >
          master
        </span>
         
        branch.
      </span>
    </div>
    <div
      class="buttonContainer"
    >
      <div
        class="buttonWrapper"
      >
        <button
          class="className button mini"
        >
          Switch Branch
        </button>
        <div
          class="base branchSwitcher"
        >
          <div
            class="background"
          />
          <div
            class="content"
          >
            <div
              class="header"
            >
              <span
                class="title"
              >
                Switch Branch
              </span>
              <div
                class="close"
              >
                <kbd
                  class="base variable dark mini"
                >
                  esc
                </kbd>
                <button
                  class="closeButton"
                >
                  <svg
                    class="lucide lucide-x"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <line
                      x1="18"
                      x2="6"
                      y1="6"
                      y2="18"
                    />
                    <line
                      x1="6"
                      x2="18"
                      y1="6"
                      y2="18"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="searchContainer"
            >
              <div
                class="base dark"
                style="width: 100%;"
              >
                <div
                  class="inputWrapper searchInput"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      class="lucide lucide-search"
                      fill="none"
                      height="18"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="11"
                        cy="11"
                        r="8"
                      />
                      <line
                        x1="21"
                        x2="16.65"
                        y1="21"
                        y2="16.65"
                      />
                    </svg>
                  </div>
                  <input
                    class="input withIcon"
                    placeholder="Type to search..."
                    type="text"
                    value=""
                  />
                  <div
                    class="loadingWrapper"
                  >
                    <div
                      class="loading center tiny"
                    >
                      <img
                        alt="A spinner indicating an ongoing operation"
                        class="spinner"
                        data-nimg="1"
                        decoding="async"
                        height="15"
                        loading="lazy"
                        src="/icons/spinner-dark.svg"
                        srcset="/icons/spinner-dark.svg 1x, /icons/spinner-dark.svg 2x"
                        style="color: transparent;"
                        width="15"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="itemsContainer"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
